modules/sbr/fsw/src/cfe_sbr_route_unsorted.c

```diff
@@ -0,0 +1,210 @@
+    memset(&CFE_SBR_RDATA, 0, sizeof(CFE_SBR_RDATA));
+
+    /* Only non-zero value for shared data initialization is the invalid MsgId */
+    for (routeidx = 0; routeidx < CFE_PLATFORM_SB_MAX_MSG_IDS; routeidx++)
+    {
+        CFE_SBR_RDATA.RoutingTbl[routeidx].MsgId = CFE_SB_INVALID_MSG_ID;
+    }
+
+    /* Initialize map */
+    CFE_SBR_Init_Map();
+}
+
+/******************************************************************************
+ *  Interface function - see API for description
+ */
+CFE_SBR_RouteId_t CFE_SBR_AddRoute(CFE_SB_MsgId_t MsgId, uint32 *CollisionsPtr)
+{
+    CFE_SBR_RouteId_t routeid    = CFE_SBR_INVALID_ROUTE_ID;
+    uint32            collisions = 0;
+
+    if (CFE_SB_IsValidMsgId(MsgId) && (CFE_SBR_RDATA.RouteIdxTop < CFE_PLATFORM_SB_MAX_MSG_IDS))
+    {
+        routeid    = CFE_SBR_ValueToRouteId(CFE_SBR_RDATA.RouteIdxTop);
+        collisions = CFE_SBR_SetRouteId(MsgId, routeid);
+
+        CFE_SBR_RDATA.RoutingTbl[CFE_SBR_RDATA.RouteIdxTop].MsgId = MsgId;
+        CFE_SBR_RDATA.RouteIdxTop++;
+    }
+
+    if (CollisionsPtr != NULL)
+    {
+        *CollisionsPtr = collisions;
+    }
+
+    return routeid;
+}
+
+/******************************************************************************
+ *  Interface function - see API for description
+ */
+CFE_SB_MsgId_t CFE_SBR_GetMsgId(CFE_SBR_RouteId_t RouteId)
+{
+    CFE_SB_MsgId_t msgid = CFE_SB_INVALID_MSG_ID;
+
+    if (CFE_SBR_IsValidRouteId(RouteId))
+    {
+        msgid = CFE_SBR_RDATA.RoutingTbl[CFE_SBR_RouteIdToValue(RouteId)].MsgId;
+    }
+
+    return msgid;
+}
+
+/******************************************************************************
+ *  Interface function - see API for description
+ */
+CFE_SB_DestinationD_t *CFE_SBR_GetDestListHeadPtr(CFE_SBR_RouteId_t RouteId)
+{
+
+    CFE_SB_DestinationD_t *destptr = NULL;
+
+    if (CFE_SBR_IsValidRouteId(RouteId))
+    {
+        destptr = CFE_SBR_RDATA.RoutingTbl[CFE_SBR_RouteIdToValue(RouteId)].ListHeadPtr;
+    }
+
+    return destptr;
+}
+
+/******************************************************************************
+ *  Interface function - see API for description
+ */
+void CFE_SBR_SetDestListHeadPtr(CFE_SBR_RouteId_t RouteId, CFE_SB_DestinationD_t *DestPtr)
+{
+
+    if (CFE_SBR_IsValidRouteId(RouteId))
+    {
+        CFE_SBR_RDATA.RoutingTbl[CFE_SBR_RouteIdToValue(RouteId)].ListHeadPtr = DestPtr;
+    }
+}
+
+/******************************************************************************
+ *  Interface function - see API for description
+ */
+void CFE_SBR_IncrementSequenceCounter(CFE_SBR_RouteId_t RouteId)
+{
+    if (CFE_SBR_IsValidRouteId(RouteId))
+    {
+        CFE_SBR_RDATA.RoutingTbl[CFE_SBR_RouteIdToValue(RouteId)].SeqCnt++;
+    }
+}
+
+/******************************************************************************
+ *  Interface function - see API for description
+ */
+CFE_MSG_SequenceCount_t CFE_SBR_GetSequenceCounter(CFE_SBR_RouteId_t RouteId)
+{
+    uint32 seqcnt = 0;
```

Comment:
Why not use a CFE_MSG_SequenceCount_t here?

Reply:
https://github.com/nasa/cFE/issues/1476